pallets/tendermint-light-client/src/lib.rs

```diff
@@ -0,0 +1,293 @@
+                now,
+            );
+
+            match verdict {
+                tendermint_light_client_verifier::Verdict::Success => {
+                    // update storage
+                    insert_light_block::<T>(hash, untrusted_block);
+                    log::info!(target: "runtime::tendermint-lc", "Successfully verified light block {:#?}", &hash);
+                    Self::deposit_event(Event::ImportedLightBlock(who, hash));
+                    Ok(())
+                }
+                tendermint_light_client_verifier::Verdict::NotEnoughTrust(voting_power_tally) => {
+                    log::warn!(target: "runtime::tendermint-lc", "Not enough voting power to accept the light block {:#?}, vote tally  {}", &hash, &voting_power_tally);
+                    fail!(<Error<T>>::NotEnoughTrust)
+                }
+                tendermint_light_client_verifier::Verdict::Invalid(why) => {
+                    log::warn!(target: "runtime::tendermint-lc", "Rejecting invalid light block {:#?} becasue {}", &hash, &why);
+
+                    fail!(<Error<T>>::InvalidBlock)
+                }
+            }
+        }
+
+        // TODO: This method will need to be called by the pallet itself if it detects a fork.
+        /// Halt or resume all light client operations
+        ///
+        /// Can only be called by root
+        #[pallet::weight((T::DbWeight::get().reads_writes(1, 1), DispatchClass::Operational))]
+        pub fn set_halted(origin: OriginFor<T>, halted: bool) -> DispatchResult {
+            ensure_root(origin)?;
+            <IsHalted<T>>::put(halted);
+
+            if halted {
+                log::warn!(target: "runtime::tendermint-lc", "Halting light client operations");
+                Self::deposit_event(Event::LightClientHalted);
+            } else {
+                log::warn!(target: "runtime::tendermint-lc", "Resuming light client operations");
+                Self::deposit_event(Event::LightClientResumed);
+            }
+
+            Ok(())
+        }
+    }
+
+    fn verify_light_block(
+        verifier: ProdVerifier,
+        untrusted_block: LightBlockStorage,
+        trusted_block: LightBlockStorage,
+        options: &Options,
+        now: Time,
+    ) -> tendermint_light_client_verifier::Verdict {
+        let untrusted_block: LightBlock = untrusted_block
+            .try_into()
+            .expect("Unexpected failure when casting untrusted block as tendermint::LightBlock");
+
+        let trusted_block: LightBlock = trusted_block
+            .try_into()
+            .expect("Unexpected failure when casting trusted block as tendermint::LightBlock");
+
+        // verify against known state
+        verifier.verify(
+            untrusted_block.as_untrusted_state(),
+            trusted_block.as_trusted_state(),
+            options,
+            now,
+        )
+    }
+
+    /// update light client storage
+    /// should only be called by a trusted origin, *after* performing a verification
```

Comment:
You call `insert_light_block` in `update_client` which only ensures that the xt wast signed. I think this is fine so only this comment needs to be corrected.

Reply:
not sure I understand the comment, you want to see the second comments line removed?